packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -130,16 +242,21 @@ class ModalBarrier extends StatelessWidget {
         excluding: !semanticsDismissible || !modalBarrierSemanticsDismissible,
         child: _ModalBarrierGestureDetector(
           onDismiss: handleDismiss,
-          child: Semantics(
-            label: semanticsDismissible ? semanticsLabel : null,
-            onDismiss: semanticsDismissible ? handleDismiss : null,
-            textDirection: semanticsDismissible && semanticsLabel != null ? Directionality.of(context) : null,
-            child: MouseRegion(
-              cursor: SystemMouseCursors.basic,
-              child: ConstrainedBox(
+          child: SemanticsClipper(
+            clipDetailsNotifier: clipDetailsNotifier,
+            child: Semantics(
+              onTapHint: semanticsOnTapHint,
```

Comment:
this and the line below should check for semanticsDismissible?

Reply:
Switched to optionally wrapping the barrier with clipper based on semantics Dismissible and whether the Notifier is null, please take a look, thanks!